Apps/phradar/radar_connector.py

```diff
@@ -0,0 +1,451 @@
+
+    def _get_system_settings(self):
+        self.save_progress("Get system settings")
+
+        url = f"{self.get_phantom_base_url()}rest/system_settings"
+        try:
+            resp = requests.get(url, verify=False)
+            resp_json = resp.json()
+        except Exception as ex:
+            self.debug_print(f"Get system settings during action: {self.get_action_identifier()}. Error:", ex)
+
+        return resp_json
+
+    def _localize_and_format_timestr(self, timestr):
+        time_zone = self.get_config().get("time_zone", "UTC")
+        parsed_datetime = parser.parse(timestr)
+        formatted_datetime = parsed_datetime.astimezone(pytz.timezone(time_zone)).strftime("%Y-%m-%d %H:%M:%S")
+        return f"{formatted_datetime} / {time_zone}"
+
+    def _format_timestr(self, timestr):
+        time_zone = self.get_config().get("time_zone", "UTC")
+        formatted_datetime = parser.parse(timestr).strftime("%Y-%m-%d %H:%M:%S")
+        return f"{formatted_datetime} / {time_zone}"
+
+    def _handle_add_note(self, param):
+        self.save_progress(f"Run action {self.get_action_identifier()}")
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        content = param.get("content")
+        category = "Splunk Phantom"
+        incident_id = param.get("incident_id")
+
+        body = {
+            "content": content,
+            "category": category,
+        }
+
+        # make rest call
+        radar_val, data = self._make_rest_call(f"/incidents/{incident_id}/notes", action_result, method="post", data=json.dumps(body))
+
+        if phantom.is_fail(radar_val):
+            return action_result.get_status()
+
+        # construct note output with fields we want to render
+        note = dict()
+        note["incident_id"] = incident_id
+        note["id"] = data["headers"]["Location"].split("/")[4]
+        note["content"] = content
+
+        self.save_progress("Add data to action result")
+        action_result.add_data(note)
+
+    def _handle_get_notes(self, param):
+        self.save_progress(f"Run action {self.get_action_identifier()}")
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        incident_id = param.get("incident_id")
+
+        # make rest call
+        radar_val, data = self._make_rest_call(f"/incidents/{incident_id}/notes", action_result, method="get")
+
+        if phantom.is_fail(radar_val):
+            return action_result.get_status()
+
+        notes = []
+
+        for note_data in data:
+            if note_data["category"] == "Splunk Phantom":
+                # construct note output with fields we want to render
+                notes.append({
+                    "incident_id": incident_id,
+                    "id": note_data["id"],
```

Comment:
use a .get() with a default value or add a try catch block here.  I believe this could cause an error if not_data['id'] does not exist.  Same for the other values below.